tests/benchmarks/cpu/masks.cpp

```diff
@@ -0,0 +1,367 @@
+            }
+        }
+    }
+
+#ifdef DETRAY_BENCHMARK_PRINTOUTS
+    constexpr detray::scalar area{4.f * r[0] * r[1]};
+    constexpr detray::scalar rest{world * world - area};
+    std::cout << "Rectangle : Inside/outside ... " << inside << " / " << outside
+              << " = "
+              << static_cast<scalar>(inside) / static_cast<scalar>(outside)
+              << " (theoretical = " << area / rest << ") " << std::endl;
+#endif  // DETRAY_BENCHMARK_PRINTOUTS
+}
+
+BENCHMARK(BM_RECTANGLE_2D_MASK)
+#ifdef DETRAY_BENCHMARK_MULTITHREAD
+    ->ThreadRange(1, benchmark::CPUInfo::Get().num_cpus)
+#endif
+    ->Unit(benchmark::kMillisecond);
+
+// This runs a benchmark on a trapezoid2D mask
+static void BM_TRAPEZOID_2D_MASK(benchmark::State &state) {
+
+    using mask_type = detray::mask<detray::trapezoid2D<>>;
+    constexpr mask_type t{0u, 2.f, 3.f, 4.f};
+
+    constexpr detray::scalar world{10.f};
+
+    constexpr detray::scalar sx{world / steps_x3};
+    constexpr detray::scalar sy{world / steps_y3};
+    constexpr detray::scalar sz{world / steps_z3};
+
+    unsigned long inside = 0u;
+    unsigned long outside = 0u;
+
+    for (auto _ : state) {
+        for (unsigned int ix = 0u; ix < steps_x3; ++ix) {
+            detray::scalar x{-0.5f * world +
+                             static_cast<detray::scalar>(ix) * sx};
+            for (unsigned int iy = 0u; iy < steps_y3; ++iy) {
+                detray::scalar y{-0.5f * world +
+                                 static_cast<detray::scalar>(iy) * sy};
+                for (unsigned int iz = 0u; iz < steps_z3; ++iz) {
+                    detray::scalar z{-0.5f * world +
+                                     static_cast<detray::scalar>(iz) * sz};
+
+                    benchmark::DoNotOptimize(inside);
+                    benchmark::DoNotOptimize(outside);
+                    mask_type::loc_point_t loc_p{
+                        t.to_local_frame(trf, {x, y, z})};
+                    if (t.is_inside(loc_p) ==
+                        detray::intersection::status::e_inside) {
+                        ++inside;
+                    } else {
+                        ++outside;
+                    }
+                }
+            }
+        }
+    }
+
+#ifdef DETRAY_BENCHMARK_PRINTOUTS
+    constexpr detray::scalar area{2.f * (t[0] + t[1]) * t[2]};
+    constexpr detray::scalar rest{world * world - area};
+    std::cout << "Trapezoid : Inside/outside ..." << inside << " / " << outside
+              << " = "
+              << static_cast<scalar>(inside) / static_cast<scalar>(outside)
+              << " (theoretical = " << area / rest << ") " << std::endl;
+#endif  // DETRAY_BENCHMARK_PRINTOUTS
+}
+
+BENCHMARK(BM_TRAPEZOID_2D_MASK)
+#ifdef DETRAY_BENCHMARKS_MULTITHREAD
+    ->ThreadRange(1, benchmark::CPUInfo::Get().num_cpus)
+#endif
+    ->Unit(benchmark::kMillisecond);
+
+// This runs a benchmark on a ring2D mask (as disc)
+static void BM_RING_2D_MASK(benchmark::State &state) {
```

Comment:
I actually think that `BM_RING_2D_MASK` and `BM_DISC_2D_MASK` are mixed up. :thinking: (I think the naming should be the other way around.) But kept this naming at first, to discuss with you before I would change it.

Reply:
Okay, I switched the function names. :wink: